lib/nebulex/adapters/local/generation.ex

```diff
@@ -242,19 +273,37 @@ defmodule Nebulex.Adapters.Local.Generation do
     # Initial state
     state = struct(__MODULE__, parse_opts(opts))
 
-    # Init cleanup timer
-    cleanup_ref =
-      if state.max_size || state.allocated_memory,
-        do: start_timer(state.gc_cleanup_max_timeout, nil, :cleanup)
+    old_strategy? =
+      state.generation_max_size == nil and state.generation_allocated_memory == nil and
+        state.generation_start_timeout == nil
 
-    # Timer ref
-    {:ok, ref} =
-      if state.gc_interval,
-        do: {new_gen(state), start_timer(state.gc_interval)},
-        else: {new_gen(state), nil}
+    new_gen(state)
 
-    # Update state
-    state = %{state | gc_cleanup_ref: cleanup_ref, gc_heartbeat_ref: ref}
+    state =
+      if old_strategy? do
+        # Init cleanup timer
+        cleanup_ref =
+          if state.max_size || state.allocated_memory,
+            do: start_timer(state.gc_cleanup_max_timeout, nil, :cleanup)
+
+        # Timer ref
+        ref =
+          if state.gc_interval,
+            do: start_timer(state.gc_interval),
+            else: nil
+
+        %{state | gc_cleanup_ref: cleanup_ref, gc_heartbeat_ref: ref}
+      else
+        ref = start_timer(state.generation_cleanup_timeout, nil, :generation_cleanup)
```

Comment:
I see you are only starting the cleanup timer, but you have to start the `:generation_start_timeout` too, which is analogous to `:gc_interval` (it should behave exactly like `:gc_interval`), and the idea is it is a periodic timer for pushing a new generation every time it runs. The creation of a new generation is not only when the size/memory cleanup runs but also when this timer occurs, it is how the generational cache works, so both timers, especially this one, are needed.

Reply:
In the handler of `:generation_cleanup`, it will check whether new generation should be created.
```size > generation_max_size or memory > generation_allocated_memory or
            now - generation_created_at > generation_start_timeout```
So only one timer is needed.

Reply:
If a generation live for more than `: generation_start_timeout`, the timer of `:generation_cleanup` will check that and create a new generation as the above code showed